File: refactoring-practice-1/src/main/java/dojo/move_method/Account.java
package dojo.move_method;

public class Account {

	private final AccountType type;
	private int daysOverdrawn;
	
	public Account(AccountType type) {
		super();
		this.type = type;
	}
	
	public int getDaysOverdrawn() {
		return daysOverdrawn;
	}
	
	public void setDaysOverdrawn(int daysOverdrawn) {
		this.daysOverdrawn = daysOverdrawn;
	}

	/*
	 * Ezt a metódust helyezzük át az AccountType osztályba.
	 */
	double getOverdraftCharge() {
		if (type.isPremium()) {
			double result = 10.0;
			if (daysOverdrawn > 7) {
				result += (daysOverdrawn - 7) + 0.85;
			}
			return result;
		} else {
			return daysOverdrawn * 1.75;
		}
	}
	
	public double getBankCharge() {
		double result = 4.5;
		if (daysOverdrawn > 0) {
			result += getOverdraftCharge();
		}
		return result;
	}
}

File: refactoring-practice-1/src/main/typescript/move_method/account.ts
import {AccountType} from "./account-type.js";

export class Account {

    private daysOverdrawn: number;

    public constructor(private type: AccountType) {
    }

    public getDaysOverdrawn() {
        return this.daysOverdrawn;
    }

    public setDaysOverdrawn(daysOverdrawn: number) {
        this.daysOverdrawn = daysOverdrawn;
    }

    /*
     * Ezt a metódust helyezzük át az AccountType osztályba.
     */
    getOverdraftCharge(): number {
        if (this.type.isPremium()) {
            let result = 10.0;
            if (this.daysOverdrawn > 7) {
                result += (this.daysOverdrawn - 7) + 0.85;
            }
            return result;
        } else {
            return this.daysOverdrawn * 1.75;
        }
    }

    public getBankCharge(): number {
        let result = 4.5;
        if (this.daysOverdrawn > 0) {
            result += this.getOverdraftCharge();
        }
        return result;
    }
}
